src/request/keyspace.rs

```diff
@@ -149,6 +149,15 @@ impl TruncateKeyspace for Vec<KvPair> {
     }
 }
 
+impl TruncateKeyspace for Vec<crate::proto::kvrpcpb::LockInfo> {
+    fn truncate_keyspace(mut self, keyspace: Keyspace) -> Self {
+        for lock in self.locks.iter_mut() {
```

Comment:
Fields of `primary_lock` & `secondaries` may also need to be truncated keyspace.